modules/c++/six.sicd/source/Utilities.cpp

```diff
@@ -130,6 +130,166 @@ Utilities::getProjectionModel(const ComplexData* data,
     }
 }
 
+six::NITFReadControl six::sicd::Utilities::getLoadedNITFReadControl(
+        const std::string& sicdPathname,
+        const std::vector<std::string>& schemaPaths)
+{
+    six::XMLControlRegistry xmlRegistry;
+    xmlRegistry.addCreator(six::DataType::COMPLEX,
+            new six::XMLControlCreatorT<
+            six::sicd::ComplexXMLControl>());
+
+    six::NITFReadControl reader;
+    reader.setXMLControlRegistry(&xmlRegistry);
```

Comment:
For now, I'd like to avoid using the singleton factory when possible (for potential thread safety reasons as well as my own pedantic hatred of singletons).  Originally the intent of the XML control stuff was to allow users to register their own XML controls, but in real life no one does this and as you mentioned, CPHD never dealt with any of this hoopla (and works just fine).  So, rather than going the canParse() route, I think we just live with the current XML control stuff for now, trying to avoid singleton use when practical, and eventually (New Year's resolution) cut it out altogether.